MyCards/MyCards/PreviewView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  PreviewView.swift
+//  MyCards
+//
+//  Created by Maciej Piotrowski on 23/11/16.
+//  Copyright © 2016 Maciej Piotrowski. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+
+class PreviewView: UIView {
+
+    let captureButton = PhotoCameraButton(frame: .zero)
+    let closeButton = CloseButton(frame: .zero)
+    let outline: UIView = UIView(frame: .zero).with { view in
+        view.layer.cornerRadius = 10
+        view.layer.borderWidth = 2
+        view.layer.borderColor = UIColor.white.cgColor
+    }
+    var controlsAlpha: CGFloat = 1.0 {
+        didSet {
+            captureButton.alpha = controlsAlpha
+            closeButton.alpha = controlsAlpha
+            outline.alpha = controlsAlpha
+        }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        videoPreviewLayer.videoGravity = AVLayerVideoGravityResizeAspectFill
+        addSubview(outline)
+        addSubview(captureButton)
+        addSubview(closeButton)
+        configureConstraints()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    var videoPreviewLayer: AVCaptureVideoPreviewLayer {
+        //swiftlint:disable force_cast
+        return layer as! AVCaptureVideoPreviewLayer
+        //swiftlint:enable force_cast
+    }
+
+    var session: AVCaptureSession? {
+        get {
+            return videoPreviewLayer.session
+        }
+        set {
+            videoPreviewLayer.session = newValue
+        }
+    }
+
+    override class var layerClass: AnyClass {
+        return AVCaptureVideoPreviewLayer.self
+    }
+
+    fileprivate func configureConstraints() {
+
+        subviews.forEach { $0.translatesAutoresizingMaskIntoConstraints = false }
+
+        let views: [String: Any] = [
+            "photoButton": captureButton,
+            "closeButton": closeButton,
+            "outline": outline,
```

Comment:
I'm leaving extra commas to every dictionary/array, because it's easier to copy paste code from previous line and rewrite it or/and I don't like adding this coma every time I decide to add a value to this struct.